src/data/nat/prime.lean

```diff
@@ -1019,7 +1042,35 @@ begin
   { simp [(coprime_zero_left _).mp hab] },
   rcases b.eq_zero_or_pos with rfl | hb,
   { simp [(coprime_zero_right _).mp hab] },
-  rw [mem_factors_mul_of_pos ha hb p, list.mem_union]
+  rw [mem_factors_mul_of_pos ha hb p, list.mem_union],
 end
 
+/-- If `x` is a prime factor of `a` then `x` is also a prime factor of `a * b` for any `b > 0` -/
+lemma mem_factors_mul_left {x a b : ℕ} (hax : x ∈ a.factors) (hb : 0 < b) : x ∈ (a*b).factors :=
+begin
+  rcases a.eq_zero_or_pos with rfl | ha,
+  { rw nat.factors_zero at hax, exact absurd hax (list.not_mem_nil x) },
+  { rw mem_factors ha at hax,
+    exact (mem_factors (mul_pos ha hb)).mpr ⟨hax.1, dvd_mul_of_dvd_left hax.2 b⟩ }
+end
+
+/-- If `x` is a prime factor of `b` then `x` is also a prime factor of `a * b` for any `a > 0` -/
+lemma mem_factors_mul_right {x a b : ℕ} (hbx : x ∈ b.factors) (ha : 0 < a) : x ∈ (a*b).factors :=
+by { rw mul_comm, exact mem_factors_mul_left hbx ha }
+
+/-- If `x` is a prime factor of `a` then the power of `x` in `a` (i.e. the greatest `k` such that
+`x^k` divides `a`) is the same as the power of `x` in `a * b`, for any `b` coprime to `a`. -/
+lemma factor_count_eq_of_coprime_left {x a b : ℕ} (hab : coprime a b) (hxa : x ∈ a.factors) :
+  list.count x a.factors = list.count x (a * b).factors :=
```

Comment:
It might be useful to `open` namespace `list` globally in the file, so that you can remove most `list` prefixes.